src/entry_point.py

```diff
@@ -201,6 +256,314 @@ def _constructor_menuinst(prefix, pkg_names=None, root_prefix=None, remove=False
+                return
+            if file.is_dir():
+                rmtree(file)
+            elif file.is_symlink() or file.is_file():
+                file.unlink()
+        except PermissionError:
+            pass
+
+    def _remove_file_and_parents(file: Path, stop_at: Path | None = None):
+        """
+        Remove a file and its parents if empty until reaching the stop_at directory.
+        """
+        if stop_at and not _is_subdir(file, stop_at):
+            raise RuntimeError(f"{file} must be a subdirectory of {stop_at}.")
+        if not file.exists():
+            return
+        _remove_file_directory(file)
+        parent = file.parent
+        while (
+            parent != stop_at and parent.is_dir() and not next(parent.iterdir(), None)
+        ):
+            _remove_file_directory(parent)
+            parent = parent.parent
+
+    def _remove_config_file_and_parents(file: Path):
+        """
+        Remove a configuration file and empty parent directories.
+
+        Only remove the configuration files created by conda.
+        For that reason, search only for specific subdirectories
+        and search backwards to be conservative about what is deleted.
+        """
+        rootdir = None
+        parts_inverse = list(reversed(file.parts))
+        for config_dir in (".config", ".conda", "conda", "xonsh"):
+            try:
+                root_index = parts_inverse.index(config_dir) + 1
+                rootdir = Path(*file.parts[:-root_index])
+                break
+            except ValueError:
+                pass
+        if not rootdir:
+            _remove_file_directory(file)
+        else:
+            _remove_file_and_parents(file, stop_at=rootdir)
+
+    print(f"Uninstalling conda installation in {uninstall_prefix}...")
+    prefixes = [
+        file.parent.parent.resolve()
+        for file in uninstall_prefix.glob(f"**/{PREFIX_MAGIC_FILE}")
+    ]
+    # Sort by path depth. This will place the root prefix first
+    # Since it is more likely that profiles contain the root prefix,
+    # this makes loops more efficient.
+    prefixes.sort(key=lambda x: len(x.parts))
+
+    # Run conda --init reverse for the shells
+    # that contain a prefix that is being uninstalled
+    anaconda_prompt = False
+    print("Running conda init --reverse...")
+    for for_user in (True, False):
+        # Run user and system reversal separately because user
+        # and system files may contain separate paths.
+        for_system = not for_user
+        anaconda_prompt = False
+        plan = _get_init_reverse_plan(
+            uninstall_prefix, prefixes, for_user, for_system, anaconda_prompt
+        )
+        # Do not call conda.core.initialize() because it will always run make_install_plan.
+        # That function will search for activation scripts in sys.prefix which do no exist
+        # in the extraction directory of conda-standalone.
+        run_plan(plan)
+        run_plan_elevated(plan)
```

Comment:
As far as I know, that's correct. There doesn't seem to be a "failed" state in the `Results` enum either.

I am printing the plan results at least so that users can check if they don't see the results they expect.